src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/AzureSqlVMCmdletBase.cs

```diff
@@ -0,0 +1,98 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Model;
+using Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Adapter;
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.SqlVirtualMachine.Common;
+using System.Management.Automation;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Common.Compute.Version_2018_04;
+using Microsoft.Azure.Commands.Common.Compute.Version_2018_04.Models;
+
+namespace Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Cmdlet
+{
+    /// <summary>
+    /// Generic cmdlet for operations that run over a sql virtual machine. 
+    /// </summary>
+    public abstract class AzureSqlVMCmdletBase : AzureSqlVirtualMachineCmdletBase<IEnumerable<AzureSqlVMModel>, AzureSqlVMAdapter>
+    {
+        /// <summary>
+        /// Intializes the model adapter
+        /// </summary>
+        /// <returns>The Sql virtual machine adapter</returns>
+        protected override AzureSqlVMAdapter InitModelAdapter()
+        {
+            return new AzureSqlVMAdapter(DefaultContext);
+        }
+
+        /// <summary>
+        /// Validate the resource id of a sql virtual machine
+        /// </summary>
+        /// <param name="sqlVirtualMachineResourceId">Resource id to be validated</param>
+        /// <returns>True if the resource id is valid, false otherwise</returns>
+        public bool ValidateSqlVirtualMachineId(string sqlVirtualMachineResourceId)
+        {
+            var regex = new Regex(@"/subscriptions/([^/]+)/resourcegroups/([^/]+)/providers/microsoft.sqlvirtualmachine/sqlvirtualmachines/([^/]+)");
+            return regex.IsMatch(sqlVirtualMachineResourceId.ToLower());
+        }
+
+        /// <summary>
+        /// Retrieve an existent virtual machine
+        /// </summary>
+        /// <param name="resourceGroup">Name of the resource group in which the virtual machine should be</param>
+        /// <param name="virtualMachineName">Name of the virtual machine</param>
+        /// <returns></returns>
+        public VirtualMachine RetrieveVirtualMachine(string resourceGroup, string virtualMachineName)
+        {
+            var client = AzureSession.Instance.ClientFactory.CreateArmClient<ComputeManagementClient>(DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+            try
+            {
+                return client.VirtualMachines.Get(resourceGroup, virtualMachineName);
+            }
+            catch(CloudException)
+            {
+                throw new PSArgumentException(
+               string.Format("The parameters provided do not correspond to a virtual machine."),
```

Comment:
Can we provide the parameter details in the exception as well.  Something like "There is no virtual machine named <virtualMachineName> in the resource group <resourceGroup>